code/datums/trading/traders/station/supply.dm

```diff
@@ -0,0 +1,60 @@
+//Supply station! Literally everything what's in need for every ship to get up and running
+
+/datum/trader/supply_station
+	name = "Trading Supply Station"
+	name_language = /decl/language/human/common
+	trade_flags = TRADER_MONEY | TRADER_GOODS | TRADER_WANTED_ONLY
+
+	margin = 1.1
+
+	insult_drop = 0 //dumber than ships
+	compliment_increase = 0
+
+	possible_trading_items = list(/obj/item/tank/oxygen/yellow = TRADER_THIS_TYPE,
+								/obj/item/tank/hydrogen = TRADER_THIS_TYPE,
+								/obj/machinery/portable_atmospherics/canister/air = TRADER_THIS_TYPE,
+								/obj/machinery/portable_atmospherics/canister/hydrogen = TRADER_THIS_TYPE,
+								/obj/machinery/portable_atmospherics/canister/oxygen = TRADER_THIS_TYPE,
+								/obj/machinery/portable_atmospherics/canister/carbon_dioxide = TRADER_THIS_TYPE,
+								/obj/item/stack/material/puck/mapped/uranium/ten = TRADER_THIS_TYPE,
+								/obj/item/stack/material/aerogel/mapped/deuterium/ten = TRADER_THIS_TYPE,
+								/obj/structure/reagent_dispensers/watertank = TRADER_THIS_TYPE,
+								/obj/structure/reagent_dispensers/fueltank = TRADER_THIS_TYPE,
+								/obj/item/storage/toolbox/mechanical = TRADER_THIS_TYPE,
+								/obj/item/storage/toolbox/repairs = TRADER_THIS_TYPE,
+								/obj/item/multitool = TRADER_THIS_TYPE,
+								/obj/item/gun/projectile/pistol/random = TRADER_THIS_TYPE,
+								/obj/item/ammo_magazine/pistol = TRADER_THIS_TYPE,
+								/obj/item/clothing/head/helmet/space/void/engineering = TRADER_THIS_TYPE,
+								/obj/item/clothing/suit/space/void/engineering = TRADER_THIS_TYPE,
+								/obj/item/stack/cable_coil/random = TRADER_THIS_TYPE,
+								/obj/item/cell/high = TRADER_THIS_TYPE,
+								/obj/item/storage/firstaid/regular = TRADER_THIS_TYPE,
+								/obj/item/storage/firstaid/adv = TRADER_THIS_TYPE,
+								/obj/item/chems/food/snacks/canned = TRADER_SUBTYPES_ONLY,
+								/obj/item/chems/food/snacks/canned/caviar = TRADER_BLACKLIST_ALL,
+								/obj/item/chems/food/drinks/cans/waterbottle = TRADER_THIS_TYPE
+								)
+
+	possible_wanted_items = list(/obj/item/trash = TRADER_SUBTYPES_ONLY)
+
+	speech = list("hail_generic"    = "Greetings! This is automated trading unit speaking. It's a pleasure to receive a connection from you, our dear customer. We are also participating in a trash recycling program right now.",
```

Comment:
I think it might be better to make `speech` a datum structure, and have your traders pick from a list of possible ones, then you could move the book trader, names, etc. onto the datums.

Reply:
like
``
/datum/trader_speech
hail = "hi"
``
this??